src/features/defi/providers/foxy/components/FoxyManager/Deposit/FoxyDeposit.tsx

```diff
@@ -0,0 +1,661 @@
+            .times(`1e+${asset.precision}`)
+            .decimalPlaces(0),
+          tokenContractAddress: tokenId,
+          userAddress: state.userAddress,
+          contractAddress,
+          wallet: walletState.wallet
+        }),
+        api.getGasPrice()
+      ])
+      dispatch({ type: FoxyDepositActionType.SET_TXID, payload: txid })
+      history.push(DepositPath.Status)
+
+      const transactionReceipt = await poll({
+        fn: () => api.getTxReceipt({ txid }),
+        validate: (result: TransactionReceipt) => !isNil(result),
+        interval: 15000,
+        maxAttempts: 30
+      })
+      dispatch({
+        type: FoxyDepositActionType.SET_DEPOSIT,
+        payload: {
+          txStatus: transactionReceipt.status === true ? 'success' : 'failed',
+          usedGasFee: bnOrZero(gasPrice).times(transactionReceipt.gasUsed).toFixed(0)
+        }
+      })
+    } catch (error) {
+      console.error('FoxyDeposit:handleDeposit error', error)
+      toast({
+        position: 'top-right',
+        description: translate('common.transactionFailedBody'),
+        title: translate('common.transactionFailed'),
+        status: 'error'
+      })
+    } finally {
+      dispatch({ type: FoxyDepositActionType.SET_LOADING, payload: false })
+    }
+  }
+
+  const handleViewPosition = () => {
+    browserHistory.push('/defi')
+  }
+
+  const handleCancel = () => {
+    browserHistory.goBack()
+  }
+
+  const validateCryptoAmount = (value: string) => {
+    const crypto = bnOrZero(balance).div(`1e+${asset.precision}`)
+    const _value = bnOrZero(value)
+    const hasValidBalance = crypto.gt(0) && _value.gt(0) && crypto.gte(value)
+    if (_value.isEqualTo(0)) return ''
+    return hasValidBalance || 'common.insufficientFunds'
+  }
+
+  const validateFiatAmount = (value: string) => {
+    const crypto = bnOrZero(balance).div(`1e+${asset.precision}`)
+    const fiat = crypto.times(marketData.price)
+    const _value = bnOrZero(value)
+    const hasValidBalance = fiat.gt(0) && _value.gt(0) && fiat.gte(value)
+    if (_value.isEqualTo(0)) return ''
+    return hasValidBalance || 'common.insufficientFunds'
+  }
+
+  const renderRoute = (route: { step?: number; path: string; label: string }) => {
+    // const apy = state.foxyOpportunity.metadata?.apy?.net_apy
+    const apy = '.2'
+    const annualYieldCrypto = bnOrZero(state.deposit?.cryptoAmount).times(bnOrZero(apy))
+    const annualYieldFiat = annualYieldCrypto.times(marketData.price)
+
+    let statusIcon: React.ReactElement = <ArrowForwardIcon />
+    let statusText = StatusTextEnum.pending
+    let statusBg = defaultStatusBg
+    if (state.deposit.txStatus === 'success') {
+      statusText = StatusTextEnum.success
+      statusIcon = <CheckIcon color='white' />
+      statusBg = 'green.500'
+    }
+    if (state.deposit.txStatus === 'failed') {
+      statusText = StatusTextEnum.failed
+      statusIcon = <CloseIcon color='white' />
+      statusBg = 'red.500'
+    }
```

Comment:
```suggestion
const { statusIcon, statusText, statusBg } = (() => {
    let statusIcon: React.ReactElement = <ArrowForwardIcon />
    let statusText = StatusTextEnum.pending
    let statusBg = defaultStatusBg
    if (state.deposit.txStatus === 'success') {
      statusText = StatusTextEnum.success
      statusIcon = <CheckIcon color='white' />
      statusBg = 'green.500'
    }
    if (state.deposit.txStatus === 'failed') {
      statusText = StatusTextEnum.failed
      statusIcon = <CloseIcon color='white' />
      statusBg = 'red.500'
    }
    return { statusIcon, statusText, statusBg }
})()

```

Reply:
I'll see your 16 lines @0xdef1cafe and raise you 12:

```ts
const { statusIcon, statusText, statusBg } = (() => {
  const statusIcon = (() =>
    state.deposit.txStatus === 'success' ? (
      <CheckIcon color='white' />
    ) : (
      <CloseIcon color='white' />
    ))()
  const statusText = (() =>
    state.deposit.txStatus === 'success' ? StatusTextEnum.success : StatusTextEnum.failed)()
  const statusBg = (() => (state.deposit.txStatus === 'success' ? 'green.500' : 'red.500'))()
  return { statusIcon, statusText, statusBg }
})()
```

Reply:
your response would be considered a string bet, so beards code stands :P

https://en.wiktionary.org/wiki/string_bet

Reply:
Yeah nah this is the guy:

```ts
const { statusIcon, statusText, statusBg } = (() => {
  const statusIcon = (() => {
    switch (state.deposit.txStatus) {
      case 'success':
      case 'failed':
        return <CheckIcon color='white' />
      default:
        return <ArrowForwardIcon />
    }
  })()
  const statusText = (() => {
    switch (state.deposit.txStatus) {
      case 'success':
        return StatusTextEnum.success
      case 'failed':
        return StatusTextEnum.failed
      default:
        return StatusTextEnum.pending
    }
  })()
  const statusBg = (() => {
    switch (state.deposit.txStatus) {
      case 'success':
        return 'green.500'
      case 'failed':
        return 'red.500'
      default:
        return defaultStatusBg
    }
  })()
  return { statusIcon, statusText, statusBg }
})()
```